workers/proof-manager/src/mock.rs

```diff
@@ -31,134 +43,121 @@ pub struct MockProofManager;
     /// Generate a dummy proof of `VALID WALLET CREATE`
     fn valid_wallet_create(
-        _witness: &SizedValidWalletCreateWitness,
+        witness: SizedValidWalletCreateWitness,
         statement: SizedValidWalletCreateStatement,
-    ) -> ValidWalletCreateBundle {
+    ) -> Result<ValidWalletCreateBundle, ProofManagerError> {
+        Self::check_constraints::<SizedValidWalletCreate>(&witness, &statement)?;
+
         let proof = dummy_proof();
-        Box::new(GenericValidWalletCreateBundle { statement, proof })
+        Ok(Box::new(GenericValidWalletCreateBundle { statement, proof }))
     }
 
     /// Generate a dummy proof of `VALID WALLET UPDATE`
     fn valid_wallet_update(
-        _witness: &SizedValidWalletUpdateWitness,
+        witness: SizedValidWalletUpdateWitness,
         statement: SizedValidWalletUpdateStatement,
-    ) -> ValidWalletUpdateBundle {
+    ) -> Result<ValidWalletUpdateBundle, ProofManagerError> {
+        Self::check_constraints::<SizedValidWalletUpdate>(&witness, &statement)?;
+
         let proof = dummy_proof();
-        Box::new(GenericValidWalletUpdateBundle { statement, proof })
+        Ok(Box::new(GenericValidWalletUpdateBundle { statement, proof }))
     }
 
     /// Generate a dummy proof of `VALID REBLIND`
     fn valid_reblind(
-        _witness: &SizedValidReblindWitness,
+        witness: SizedValidReblindWitness,
         statement: ValidReblindStatement,
-    ) -> ValidReblindBundle {
+    ) -> Result<ValidReblindBundle, ProofManagerError> {
+        Self::check_constraints::<SizedValidReblind>(&witness, &statement)?;
+
         let proof = dummy_proof();
-        Box::new(GenericValidReblindBundle { statement, proof })
+        Ok(Box::new(GenericValidReblindBundle { statement, proof }))
     }
 
     /// Create a dummy proof of `VALID COMMITMENTS`
     fn valid_commitments(
-        _witness: &SizedValidCommitmentsWitness,
+        witness: SizedValidCommitmentsWitness,
         statement: ValidCommitmentsStatement,
-    ) -> ValidCommitmentsBundle {
+    ) -> Result<ValidCommitmentsBundle, ProofManagerError> {
+        Self::check_constraints::<SizedValidCommitments>(&witness, &statement)?;
+
         let proof = dummy_proof();
-        Box::new(GenericValidCommitmentsBundle { statement, proof })
+        Ok(Box::new(GenericValidCommitmentsBundle { statement, proof }))
     }
 
     /// Create a dummy proof of `VALID MATCH SETTLE`
     fn valid_match_settle(
-        _witness: &SizedValidMatchSettleWitness,
+        witness: SizedValidMatchSettleWitness,
         statement: SizedValidMatchSettleStatement,
-    ) -> ValidMatchSettleBundle {
+    ) -> Result<ValidMatchSettleBundle, ProofManagerError> {
+        Self::check_constraints::<SizedValidMatchSettle>(&witness, &statement)?;
+
         let proof = dummy_proof();
-        Box::new(GenericMatchSettleBundle { statement, proof })
+        Ok(Box::new(GenericMatchSettleBundle { statement, proof }))
     }
-}
-
-#[cfg(test)]
-mod test {
```

Comment:
we don't need to test the mock, and it's out of scope now that it requires a valid witness and statement